lib/puppet/file_serving/http_metadata.rb

```diff
@@ -0,0 +1,29 @@
+require 'puppet/file_serving/metadata'
+
+# Simplified metadata representation, suitable for the information
+# that is available from HTTP headers.
+class Puppet::FileServing::HttpMetadata < Puppet::FileServing::Metadata
+
+  def initialize(http_response, path = '/dev/null')
+    super(path)
+
+    # ignore options that do not apply to HTTP metadata
+    @owner = @group = @mode = nil
+
+    if checksum = http_response['content-md5']
+      # convert base64 digest to hex
+      checksum = checksum.unpack("m0").first.unpack("H*").first
+      @checksum_type = 'md5'
```

Comment:
If the checksum type doesn't match the system config or resource's checksum type, this will always resync the resource. That's not a great behavior, but I don't immediately have a suggestion how to fix it.